Enterprise-CMCS/eAPD
# [Feature] #4415 Activities Overview for MMIS

Resolves #4415 

### Description

### Changes
#### Direct MMIS changes
- Renamed the component `Overview.js` to `HITECHOverview.js` since this is specific to HITECH types
- Added an `MMISOverview` component since there is no overlap in the fields displayed (aside from the Activity Name now handled by the `NameForm` component) between `HITECHOverview` and `MMISOverview`
    - `MMISOverview` displays these sections:
        - Activity Name
        - Activity Snapshot
        - Problem Statement
        - Proposed Solution
- Created “new” `Overview.js` to be a wrapper component
    - Houses a mapping of the APD type to the chosen overview component (`MMISOverview` or `HITECHOverview`). This will allow us to add more types in the future
    - Sets the default type to be HITECH in case existing data does not have the `apdType` field
- Added redux actions for the new MMIS sections (`activitySnapshot`, `problemStatement`, `proposedSolution`)

#### Other indirect changes
- Split `NameAndFundingSourceForm` component into two: `NameForm` and `FundingSourceForm`
    - `NameForm` is used by `MMISOverview`
    - Both `NameForm` and `FundingSourceForm` are used by `HITECHOverview`
    - Renamed `NameAndFundingSourceForm.test.js` to `NameForm.test.js` since the test file was specific to testing the name (funding tests will be needed)
- Removed the NameAndFundingSource directory
    - `NameForm` and `FundingSourceForm` will live in the overview directory
    - Removed the only other component in this directory since it was unused (`NameAndFundingSourceReview` component and associated test file)

#### Content/Yaml changes
- Created `comprehensiveOverview.yaml` for content in the Comprehensive Overview section of an MMIS type
- Added content to the existing `overview.yaml` for an MMIS type

#### Dev Notes
- Further changes can be made to decouple name and funding source in the `nameAndFundingSourceSchema` schema

### Automated test cases written

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| Activity Overview for MMIS APD Type | Viewing existing data | Data displays in correct form field | jest |
| Activity Overview for MMIS APD Type | Viewing existing data | Data displays in correct form field | cypress |
| Activity Overview for MMIS APD Type with Admin check on | Fields are blank | Validation errors appear for each field | cypress |
| Activity Overview for MMIS APD Type with Admin check on | Fields are filled in | Validation errors disappear for each field | cypress |

### Steps to manually verify this change

1. Using [the figma](https://www.figma.com/file/hJpKHKU6fz5J0Z7fisSwa2/eAPD-MMIS-2022?node-id=1388%3A12346&t=3IIABZjkZzkxAArE-0), visit the Activity Overview page of an Activity for an MMIS APD to verify the page content matches the designs.
2. Ensure all the fields allow input and save entered data.
3. Turn on admin check to verify that validation errors display if any of the fields are blank.

### This pull request is ready to code review when

      tests, refactor
      screenreader, text scaling) OR an exemption is documented

### This pull request is ready to test when


### This pull request is ready to review when the QA has


### This pull request can be merged when
